amundsen_common/models/user.py

```diff
@@ -1,29 +1,85 @@
-from typing import Optional
+from typing import Optional, Dict
 
 import attr
+from marshmallow import ValidationError, validates_schema, pre_load
 from marshmallow_annotations.ext.attrs import AttrsSchema
 
+"""
+TODO: Explore all internationalization use cases and
+redesign how User handles names
+
+TODO - Delete pre processing of the Data
+Once all of the upstream services provide a complete User object we will no
+longer need to supplement the User objects as done in `preprocess_data`
+"""
+
 
 @attr.s(auto_attribs=True, kw_only=True)
 class User:
-    # ToDo (Verdan): Make ID a required field.
+    # ToDo (Verdan): Make user_id a required field.
     #  In case if there is only email, id could be email.
     #  All the transactions and communication will be handled by ID
-    id: Optional[str] = None
+    user_id: Optional[str] = None
     email: Optional[str] = None
     first_name: Optional[str] = None
     last_name: Optional[str] = None
     full_name: Optional[str] = None
+    display_name: Optional[str] = None
     is_active: bool = True
     github_username: Optional[str] = None
     team_name: Optional[str] = None
     slack_id: Optional[str] = None
     employee_type: Optional[str] = None
     manager_fullname: Optional[str] = None
+    manager_email: Optional[str] = None
+    manager_id: Optional[str] = None
```

Comment:
btw, do you plan to use manager_id in the future?

Reply:
^^ @verdan

Reply:
@feng-tao yes we do. So same as user, any user manager related operation will be handled through the key. Not via email.